### ISSUE TITLE
stream-testkit: issue with scala 2.12 usage of expectNextWithTimeoutPF

### ISSUE THREAD

**Issue by pjfanning** [apache/pekko#1393]
see https://github.com/apache/pekko-persistence-cassandra/pull/212
there change that seems to have caused the issue is #1218

**Comment by pjfanning** [comment]
@He-Pin what do you think we should do here? Should we document the issue we see in https://github.com/apache/pekko-persistence-cassandra/pull/212 or is there a change we can make to stream-testkit to avoid the issue going forward?

**Comment by He-Pin** [comment]
I was expected to avoid a double evaluation, but that can only be done with 2.13, we may better go with 2.13 abd I expect drop the 2.12 support in 1.2.x anyway

**Comment by pjfanning** [comment]
@He-Pin This new method in https://github.com/He-Pin/incubator-pekko/blob/05248075bd13a336c9af079cb44a61986dc1f751/stream-testkit/src/main/scala/org/apache/pekko/stream/testkit/StreamTestKit.scala#L869 doesn't make sense to me. The PartialFunction is a Scala function, not a Java function. This seems like a not very useful Java API. The other changes in #1186 seem ok to me but could we consider these?
* removing the Java API expectNextWithTimeoutPF
* or changing the Java API expectNextWithTimeoutPF to use a Java function.

**Comment by He-Pin** [comment]
Yes, but a user can build a PartialFunction with the helper, such as the Receive builder.

**Comment by pjfanning** [comment]
> Yes, but a user can build a PartialFunction with the helper, such as the Receive builder.

The ReceiveBuilder will not work for this stream-testkit use case. Unless we have a builder that can build a `PartialFunction[Any, T]` - does that exist?

**Comment by He-Pin** [comment]
@pjfanning Yes, `org.apache.pekko.japi.pf.PFBuilder`.

**Comment by pjfanning** [comment]
> @pjfanning Yes, `org.apache.pekko.japi.pf.PFBuilder`.

Thanks - there seem to be some useful classes in the `org.apache.pekko.japi.pf` package.

https://pekko.apache.org/japi/pekko/current/org/apache/pekko/japi/pf/package-summary.html

**Comment by raboof** [comment]
> @He-Pin what do you think we should do here? Should we document the issue we see in [apache/pekko-persistence-cassandra#212](https://github.com/apache/pekko-persistence-cassandra/pull/212) or is there a change we can make to stream-testkit to avoid the issue going forward?

If I understand correctly this is only an issue when using Scala 2.12, right? In that case I think the note in #1403 seems sufficient and IMHO we can close this issue (or keep it open for if anyone is interested in looking into it, but remove it from the 1.1.0 milestone)
